### PR TITLE
feature (view): add openseadragon viewer component

### PR DESCRIPTION
**Pull request by aeneaskaiser**
- add StillImageOSDViewerComponent: displays an array of StillImageFileValues in with OpenSeadragon
- add instance of StillImageOSDViewerComponent to ObjectComponent

Note: after the first pull/merge of this branch you need to run 'npm install' to download the OpenSeadragon sources to your local environment.

### CODE REVIEW

File: src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          "x": i,
+          "y": 0
+        };
+      }
+      this.viewer.open(tileSources);
+    }
+
+    private getImagesToDisplay() {
+      if (this.ignorePreviews) {
+        return this.fileValues.filter( (image) => { return !image.isPreview; } );
```
Comment (tobiasschweizer) [inline src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts]: This is much nicer than the template solution :-)

File: src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          "tileSource": {
+           "@context": "http://iiif.io/api/image/2/context.json",
+           "@id": sipiBasePath,
+            "height": height,
+            "width": width,
+            "profile": [ "http://iiif.io/api/image/2/level2.json" ],
+            "protocol": "http://iiif.io/api/image",
+            "tiles": [{
+              "scaleFactors": [ 1, 2, 4, 8, 16, 32 ],
+              "width": 1024
```
Comment (tobiasschweizer) [inline src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts]: Could you provide a link to the documentation that explains what these config options mean?

File: src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        } else {
+          viewerContainer.classList.remove("fullscreen");
+        }
+      })
+    }
+
+    private openImages() {
+      let imagesToDisplay = this.getImagesToDisplay();
+      let tileSources = [];
+      for (let i = 0; i < imagesToDisplay.length; i++) {
```
Comment (aeneaskaiser) [inline src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts]: refactored in commit 3770787

File: src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        let sipiBasePath = imagesToDisplay[i].imageServerIIIFBaseURL + "/" + imagesToDisplay[i].imageFilename;
```
Comment (aeneaskaiser) [inline src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts]: I usually follow the policy 'implement the convenience method as soon as i have to write the same code again'. So far it's only used in one place. But i'm not opposed to having a convenience method

File: src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts (hunk 2 above)
Comment (aeneaskaiser) [inline src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts]: see:
http://iiif.io/api/image/2.1/#technical-properties 
https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#open

the wording in the first link is a bit ambiguous, more detailed info in the implementation notes:
http://iiif.io/api/image/2.0/#a-implementation-notes
or directly in the sources:
https://github.com/openseadragon/openseadragon/blob/cfc339edea0a65575c7904ec82e72e911c584825/src/iiiftilesource.js

File: src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            }]
+          },
+          "x": i,
+          "y": 0
+        };
+      }
+      this.viewer.open(tileSources);
+    }
+
+    private getImagesToDisplay() {
```
Comment (aeneaskaiser) [inline src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts]: refactored in commit 3770787

File: src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      let imagesToDisplay = this.getImagesToDisplay();
+      let tileSources = [];
+      for (let i = 0; i < imagesToDisplay.length; i++) {
+        let sipiBasePath = imagesToDisplay[i].imageServerIIIFBaseURL + "/" + imagesToDisplay[i].imageFilename;
+        let width = imagesToDisplay[i].dimX;
+        let height = imagesToDisplay[i].dimY;
+        tileSources[i] = {
+          "tileSource": {
+           "@context": "http://iiif.io/api/image/2/context.json",
+           "@id": sipiBasePath,
```
Comment (aeneaskaiser) [inline src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts]: No - as long as we use OpenSeadragon as above, by providing the full tileSource configurations to `viewer.open()`, OpenSeadragon will not try to dereference the id.
The example mentioned by you is just another way to provide the configuration.

**Pull request merged**